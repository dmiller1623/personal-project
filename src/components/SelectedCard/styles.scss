.selected-card {
  display: flex;
  flex-direction: column;
  background-color: #E4E4E1;
  background-image: radial-gradient(at top center, rgba(255,255,255,0.03) 0%, rgba(0,0,0,0.03) 100%), linear-gradient(to top, rgba(255,255,255,0.1) 0%, rgba(143,152,157,0.60) 100%);
  background-blend-mode: normal, multiply; 
  height: 24rem;
  width: 100%;
  margin: 1rem;
  border-radius: 15px;
  // width: 95%;
  padding: .2rem .6rem;
}

.selected-card-info {
  width: 100%;
  display: flex;
  flex-direction: row;
  // color: white;
}

.selected-card-title {
  width: 100%;
  display: flex;
  justify-content: space-between;
}

.selected-card-description {
  overflow: scroll;
  padding: .4rem;
  width: 65%;
  display: flex;
}

.selected-card-links {
  align-items: center;
  width: 35%;
  flex-direction: column;
  display: flex;
  // justify-content: flex-end;
}

.minus-button {
  margin-right: 2.2rem;
  width: 2rem;
  height: 2rem;
}

.minus-button:hover {
  cursor: pointer;
}

 .searched-title {
  margin-left: 2rem;
  color: black;
  font-weight: bold;
}

p {
  color: black;
}

