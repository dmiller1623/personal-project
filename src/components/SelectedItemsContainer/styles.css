.selected-items-container {
  overflow: scroll;
  flex-wrap: wrap;
  display: flex;
  width: 80%;
  height: 100vh;
  justify-content: center; }

.qr-link {
  text-decoration: none;
  font-size: 3rem;
  height: 4rem;
  border: 1px solid black;
  color: white;
  text-align: center; }
