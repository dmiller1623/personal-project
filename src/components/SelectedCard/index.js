import React from 'react'
import minusSvg from '../../images/minus-symbol.svg'
import './styles.css'
import ShowMoreText from 'react-show-more-text';
import YouTube from 'react-youtube';


const SelectedCard = (props) => {
  return (
    <article className='selected-card'>
      <div className='selected-card-title'>
        {/* <div className='selected-button-section'> */}
          <img src={minusSvg} className='minus-button' alt='delete button' onClick={() => props.deleteResources(props)}/>
        {/* </div> */}
        {/* <div className='selected-search-heading'> */}
          <h1 className='searched-title'>{props.Name}</h1>
        {/* </div> */}
      </div>
      <div className='selected-card-info'>
        <div className='selected-card-description'> 
          <ShowMoreText>{props.wTeaser}</ShowMoreText>
        </div>
        <div className='selected-card-links'>
          {props.yUrl &&
            <YouTube
              className='youtube-video'
              videoId={props.yUrl.slice(39, 50)}
              opts={{
                height: '300',
                width: '300',
                autoplay: 1
              }}
            />}
            <a href={props.wUrl}>Wikipedia Link</a>
        </div>
      </div> 
    </article>
  )
}

export default SelectedCard