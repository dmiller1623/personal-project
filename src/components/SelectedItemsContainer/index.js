import React from 'react'
import SelectedCard from '../SelectedCard'

const SelectedItemsContainer = (additionalResources) => {
  // const displayResources = additionalResources.map((resource, index) => {
  //   return <SelectedCard 
  //   {...resource}
  //   key={index}
  //   />
  // })

  return (
    <div>
      nskllksnx
    </div>
  )
}

export default SelectedItemsContainer