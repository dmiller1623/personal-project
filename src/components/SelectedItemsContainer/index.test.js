import React from 'react'
import { shallow } from 'enzyme'
import SelectedItemsContainer from '../SelectedItemsContainer'
import SelectedCard from '../SelectedCard'

describe('SelectedItemsContainer', () => {
  let wrapper
  let additionalResources = []
  let deleteResources = jest.fn()
  it('should match the snapshot', () => {
    wrapper = shallow(<SelectedItemsContainer additionalResources={additionalResources} deleteResources={deleteResources}/>)

    expect(wrapper).toMatchSnapshot()
  })

  it('should render the same amount of Selected Cards', () => {
    let additionalResources = [{ name: 'jim morrision' }, { name: 'the doors' }]
    wrapper = shallow(<SelectedItemsContainer additionalResources={additionalResources} deleteResources={deleteResources}/>)

    expect(wrapper.find(SelectedCard).length).toEqual(2)
  })
})