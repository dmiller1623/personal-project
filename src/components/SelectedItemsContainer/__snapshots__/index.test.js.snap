// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedItemsContainer should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SelectedItemsContainer
    additionalResources={Array []}
    deleteResources={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [],
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="qr-link"
          to="/QrcodePage"
>
          Generate Qr Code
</NavLink>,
      ],
      "className": "selected-items-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "Generate Qr Code",
          "className": "qr-link",
          "to": "/QrcodePage",
        },
        "ref": null,
        "rendered": "Generate Qr Code",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [],
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="qr-link"
            to="/QrcodePage"
>
            Generate Qr Code
</NavLink>,
        ],
        "className": "selected-items-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "Generate Qr Code",
            "className": "qr-link",
            "to": "/QrcodePage",
          },
          "ref": null,
          "rendered": "Generate Qr Code",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
